xds/internal/xdsclient/filter_chain.go

```diff
@@ -431,7 +432,25 @@ func filterChainFromProto(fc *v3listenerpb.FilterChain) (*FilterChain, error) {
 	if sc.RequireClientCert && sc.RootInstanceName == "" {
 		return nil, errors.New("security configuration on the server-side does not contain root certificate provider instance name, but require_client_cert field is set")
 	}
-	return &FilterChain{SecurityCfg: sc}, nil
+
+	var httpFilters []HTTPFilter
+	for _, filter := range fc.GetFilters() {
+		tc := filter.GetTypedConfig()
```

Comment:
Is this the same logic as https://github.com/grpc/grpc-go/blob/master/xds/internal/xdsclient/xds.go#L303? Can that code be reused?

Reply:
Actually, that code can be just moved here I believe?

It's called right before `NewFilterChainManager()` https://github.com/grpc/grpc-go/blob/master/xds/internal/xdsclient/xds.go#L275, to validate filters (but won't keep them) https://github.com/grpc/grpc-go/blob/master/xds/internal/xdsclient/xds.go#L324

I think `validateNetworkFilterChains` can be deleted?

Reply:
It's the same. I figured it makes sense to do validation, then conversion. Should I try and make validation + data conversion in the same place?